v2/internal/reconcilers/common.go

```diff
@@ -28,9 +28,17 @@ func LogObj(log logr.Logger, level int, note string, obj genruntime.MetaObject)
 	if log.V(level).Enabled() {
 		ourAnnotations := make(map[string]string)
 		for key, value := range obj.GetAnnotations() {
-			if strings.HasPrefix(key, "serviceoperator.azure.com") {
-				ourAnnotations[key] = value
+			if !strings.HasPrefix(key, "serviceoperator.azure.com") {
+				// Skip annotations that aren't ours
+				continue
 			}
+
+			if strings.Contains(key, "token") {
```

Comment:
minor: Consider making this more specific?

I think we have a const for this annotation someplace?

Reply:
I needed to move the annotations to avoid a circular import cycle to do this. Done.